core-java-security/src/main/java/com/baeldung/ssl/EnableTLSv12.java

```diff
@@ -0,0 +1,134 @@
+
+public class EnableTLSv12 {
+
+    public String url = "";
+    public Integer port = null;
+
+    public EnableTLSv12() {
+    }
+
+    public static void main(String[] args) {
+        try {
+            EnableTLSv12 enableTLSv12 = new EnableTLSv12();
+            if (args.length != 2) {
+                System.out.println("Provide the server url and the secure port:");
+                System.exit(-1);
+            }
+            enableTLSv12.getHost(args);
+            enableTLSv12.getPort(args);
+            enableTLSv12.enableTLSv12UsingHttpConnection();
+            enableTLSv12.enableTLSv12UsingProtocol();
+            enableTLSv12.enableTLSv12UsingSSLContext();
+            enableTLSv12.enableTLSv12UsingSSLParameters();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private void getPort(String[] args) {
+        url = args[0];
+    }
+
+    private void getHost(String[] args) {
+        String portNumber = args[1];
+        port = Integer.parseInt(portNumber);
+    }
+
+    private void handleCommunication(SSLSocket socket, String usedTLSProcess) throws IOException {
+        System.out.print("Enabled TLS v1.2 on " + usedTLSProcess);
+        PrintWriter out = new PrintWriter(new BufferedWriter(new OutputStreamWriter(socket.getOutputStream())));
+        out.println("GET / HTTP/1.0");
+        out.println();
+        out.flush();
+        if (out.checkError())
+            System.out.println("SSLSocketClient:  java.io.PrintWriter error");
+
+        BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+
+        String inputLine;
+        while ((inputLine = in.readLine()) != null)
+            System.out.println(inputLine);
+
+        in.close();
+        out.close();
+        socket.close();
+    }
+
+    public void enableTLSv12UsingSSLParameters() {
+        try {
+            SSLSocketFactory socketFac = (SSLSocketFactory) SSLSocketFactory.getDefault();
+            SSLSocket sslSocket = (SSLSocket) socketFac.createSocket(url.trim(), port);
+            SSLParameters params = new SSLParameters();
+            params.setProtocols(new String[] { "TLSv1.2" });
+            sslSocket.setSSLParameters(params);
+            sslSocket.startHandshake();
+            handleCommunication(sslSocket, "SSLSocketFactory-SSLParameters");
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void enableTLSv12UsingProtocol() throws IOException {
+        try {
+            SSLSocketFactory socketFac = (SSLSocketFactory) SSLSocketFactory.getDefault();
+            SSLSocket sslSocket = (SSLSocket) socketFac.createSocket(url.trim(), port);
+            sslSocket.setEnabledProtocols(new String[] { "TLSv1.2" });
+            sslSocket.startHandshake();
+            handleCommunication(sslSocket, "SSLSocketFactory-EnabledProtocols");
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Let's either throw the exception the signature or wrap this exception in something like an `IllegalStateException`. Same for the other places where you have an `e.printStackTrace`